test: Istio e2e test suite

yashpal2104 opened the pull request:
### Key Changes

- Adds a comprehensive e2e test suite targeting Istio-enabled scenarios.
- Tests common installation/uninstallation/wait and verifies istio resources and we will also need "isInstalled" + "install" + "wait" style checks to handle this istio-ingressgateway resource.
- Adds Cross-platform istioctl installation for testing locally 
- Provides a foundation for future Istio-related test expansion and more robust CI coverage.

<!-- 
⚠️ please review https://www.kubeflow.org/docs/about/contributing/

Thank you for contributing to Kubeflow!

If there are related issues, please reference them using one of the following:

 closes: #ISSUE
 related: #ISSUE

Please remember:
 - provide enough information so that others can review your pull request
 - use a semantic title for your pull request (like "fix: xxxxx" or "feat: xxxxx", see contributing guide)
 - the title of your pull request will be used to generate the changelog entry, so make it count!
-->

yashpal2104 committed:
test: implement Istio installation utilities for e2e testing

- Add InstallIstioctl() function to download and install istioctl binary
- Add InstallIstioMinimalWithIngress() to set up Istio with minimal profile
- Add IsIstioInstalled() and WaitIstioctlAvailable() helper functions
- Use positional formatting in URL template for istioctl downloads
- Support configurable Istio namespace for installation
- Add error handling and proper command output redirection

This enables e2e tests to automatically set up Istio service mesh
components required for workspace HTTP proxy functionality.

Signed-off-by: Yash Pal <yashpal2104@gmail.com>

yashpal2104 committed:
feat: add cross-platform istioctl installation and Istio gateway management

- Add InstallIstioctl() with OS/arch detection for Linux, macOS, Windows
- Add comprehensive Istio gateway functions (install, check, wait)
- Enhance WaitIstioAvailable() to verify all Istio components
- Add UninstallIstioctl() for complete cleanup
- Use platform-specific download methods with fallbacks

Signed-off-by: Yash Pal <yashpal2104@gmail.com>

yashpal2104 committed:
feat: add istioctl installation and istio deployment support for e2e tests

- Add istioctl binary download and installation for multiple platforms
- Implement Istio minimal profile deployment with ingress gateway
- Add comprehensive Istio availability checks and wait functions
- Improve error handling and cleanup for test components
- Add platform-specific binary management for Windows/Linux/macOS

Signed-off-by: Yash Pal <yashpal2104@gmail.com>

andyatmiami reviewed (changes requested):
Great start - lets get these issues explained and/or resolved first and I'll continue with review afterwards.

Send me any error logs you may have w.r.t the problems you were having with CertManager (once you undo those changes in this PR) and I can help debug.

yashpal2104 committed:
fix: Auto-install Istio in e2e tests when missing

- Add InstallIstio() function with auto-detection
- Modify WaitIstioAvailable() to install if istio-system namespace missing
- Improve Istio/istioctl installation checks
- Fix cert-manager "Stdout already set" error

Resolves "istio-system namespace not found" test failures.

Signed-off-by: Yash Pal <yashpal2104@gmail.com>

yashpal2104 committed:
add: IstioURLTemplate

Signed-off-by: Yash Pal <yashpal2104@gmail.com>

yashpal2104 committed:
remove windows compatibility

Signed-off-by: Yash Pal <yashpal2104@gmail.com>

andyatmiami reviewed (changes requested):
Really nice progress - I am now able to observe the `test-e2e` command completing successfully 💯 

```
Ran 1 of 1 Specs in 132.105 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
```

However, I think we still want to continue to iterate on this PR to keep it more focused in scope and consistent in its implementation as compared to CertManager to aid in understandability and maintainability of future contributors.
